### PR TITLE
add list containers

### PR DESCRIPTION
**Pull request by xlgao-zju**
Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/27)

### CODE REVIEW

File: pkg/hyper/hyper.go [hunk 1]
```diff
@@ -238,7 +238,73 @@ func (h *Runtime) RemoveContainer(rawContainerID string) error {
...
+			Attempt: &attempt,
+		}
+
+		containers = append(containers, &kubeapi.Container{
+			Id:          &c.ContainerID,
+			Metadata:    containerMetadata,
+			Image:       &kubeapi.ImageSpec{Image: &info.Container.Image},
+			ImageRef:    &info.Container.ImageID,
+			State:       &state,
+			Labels:      kubeletLabels,
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: Missed `CreateAt` and `PodSandboxId`

File: pkg/hyper/client.go [hunk 2]
```diff
@@ -266,3 +266,22 @@ func (c *Client) RemoveImage(image, tag string) error {
...
+
+// GetContainerList gets a list of containers
+func (c *Client) GetContainerList(auxiliary bool) ([]*types.ContainerListResult, error) {
+	ctx, cancel := getContextWithTimeout(hyperContextTimeout)
+	defer cancel()
+
+	req := types.ContainerListRequest{
+		Auxiliary: auxiliary,
+	}
+	containerList, err := c.client.ContainerList(
```
Comment (feiskyer) [inline pkg/hyper/client.go]: nit: Format params to one line since they are so short.

File: pkg/hyper/hyper.go (hunk 1 above)
Comment (xlgao-zju) [inline pkg/hyper/hyper.go]: `kubeapi.Container` dose not have `CreateAt` and `PodSandboxId`.

Comment (feiskyer) [inline pkg/hyper/hyper.go]: Hmm, it is in the latest kubernetes. Please leave a todo for this.

File: pkg/hyper/hyper.go [hunk 3]
```diff
@@ -244,7 +244,74 @@ func (h *Runtime) RemoveContainer(rawContainerID string) error {
...
+			}
+		}
+
+		containerMetadata := &kubeapi.ContainerMetadata{
+			Name:    &containerName,
+			Attempt: &attempt,
+		}
+
+		containers = append(containers, &kubeapi.Container{
+			// TODO: Add CreateAt and PodSandboxId fields when we update kubeapi to latest.
```
Comment (xlgao-zju) [inline pkg/hyper/hyper.go]: Add a TODO here.

**Comment by feiskyer** [comment]
LGTM. Thanks.

**Pull request merged**